### PR TITLE
update CI envs to use fsspec master + python 3.9

### PR DESCRIPTION
**Pull request by rabernat**
If this works, it supersedes #83.

### CODE REVIEW

**Comment by rabernat** [comment]
Well that did not go well. I canceled after 50m...

**Comment by rabernat** [comment]
This works fine on my macbook. Frustrated by CI again.

File: ci/py3.9.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  - h5py
+  - hdf5
+  - lxml    # Optional dep of pydap
+  - netcdf4
+  - numpy
+  - pandas
+  - pip
+  - prefect
+  - pydap
+  - pynio
```
Comment (andersy005) [inline ci/py3.9.yml:23]: As far I can tell, the latest build of pynio on conda-forge supports `Python <=3.8`. Remove it from the environment fixes the issue (tested on a linux system). 

```suggestion
```

**Comment by rabernat** [comment]
Yay! One test successfully finished. The other two are hung I think.

**Pull request merged**